core/src/main/scala/kafka/server/SharedServer.scala

```diff
@@ -94,6 +95,7 @@ class SharedServer(
   val time: Time,
   private val _metrics: Metrics,
   val controllerQuorumVotersFuture: CompletableFuture[JMap[Integer, InetSocketAddress]],
+  val bootstrapServers: JCollection[InetSocketAddress],
```

Comment:
I don't see why this needs to be a new argument. We already have KafkaConfig passed in as an argument, and this just comes from `QuorumConfig.parseBootstrapServers(config.quorumBootstrapServers)`.

Adding a new argument here creates a lot of churn, which is a lot of work for merges and such.

Reply:
It's the same reason why the `controllerQuorumVotersFuture` is passed as an argument and it is not read from the `KafkaConfig`. Tests tend to configure the voters using the `0` port so that the OS can dynamically assign an open port. We need the same functionality for `bootstrapServers` in most tests the bootstrap server are the same se of endpoint as the voters but without the node id.

Having said that I have plans to remove all of these complexity after 3.8. This is mainly here before of kafka raft cluster test kit. I created this issue to track this work: https://issues.apache.org/jira/browse/KAFKA-16653

Reply:
ok.